volumes/repository.go

```diff
@@ -80,38 +81,36 @@ func (r *Repository) newVolume(path string, writable bool) (*Volume, error) {
 }
 
 func (r *Repository) restore() error {
+	log.Debugf("Restoring volumes")
 	dir, err := ioutil.ReadDir(r.configPath)
 	if err != nil {
 		return err
 	}
 
-	for _, v := range dir {
-		id := v.Name()
-		path, err := r.driver.Get(id, "")
+	for _, d := range dir {
+		f, err := os.Open(filepath.Join(r.configPath, d.Name(), "config.json"))
 		if err != nil {
-			log.Debugf("Could not find volume for %s: %v", id, err)
-			continue
+			log.Debugf("error reading volume config: %v", err)
 		}
-		vol := &Volume{
-			ID:         id,
-			configPath: r.configPath + "/" + id,
+		dec := json.NewDecoder(f)
+		v := &Volume{
+			configPath: filepath.Join(r.configPath, d.Name()),
 			containers: make(map[string]struct{}),
-			Path:       path,
 		}
-		if err := vol.FromDisk(); err != nil {
```

Comment:
I see you are not using any of these methods now like `FromDisk`.  Is this not used at all because if not, you should delete it.